src/gt4py/utils/meta.py

```diff
@@ -244,6 +245,18 @@ def generic_visit(self, node, **kwargs):
         return node
 
 
+def get_qualified_name(base_node: Union[ast.Name, ast.Attribute]):
+    node = base_node.body if isinstance(base_node, ast.Expression) else base_node
+    if isinstance(node, ast.Name):
+        return node.id
+    elif isinstance(node, ast.Attribute):
+        return get_qualified_name(node.value) + "." + node.attr
+    else:
+        raise ValueError(
+            f"Expected a node of type Union[ast.Name, ast.Attribute], but got {type(node)}"
+        )
```

Comment:
The missing underscore is certainly my bad... Might be a relict from C that I don't like this style of having an else for the fail safe case, but fine with me.

```suggestion
def _get_qualified_name(base_node: Union[ast.Name, ast.Attribute]):
    node = base_node.body if isinstance(base_node, ast.Expression) else base_node
    if isinstance(node, ast.Name):
        return node.id
    elif isinstance(node, ast.Attribute):
        return _get_qualified_name(node.value) + "." + node.attr
    else:
        raise ValueError(
            f"Expected a node of type Union[ast.Name, ast.Attribute], but got {type(node)}"
        )
```